### PR TITLE
CI: add vale linter

### PR DESCRIPTION
**Pull request by ran-huang**
Signed-off-by: Ran <huangran@pingcap.com>

### What is changed, added, or deleted? (Required)

Add vale linter and enable 14 rules at error level.

### Which TiDB Operator version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v1.3 (TiDB Operator 1.3 versions)
- [x] v1.2 (TiDB Operator 1.2 versions)
- [ ] v1.1 (TiDB Operator 1.1 versions)
- [ ] v1.0 (TiDB Operator 1.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: 
- Other reference link(s):

### CODE REVIEW

File: .vale.ini [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+IgnoredScopes = code, tt
+SkippedScopes = pre, figure, blockquote, script, style
+
+[*.md]
+#TokenIgnores = (<[^<>\n]+>), \
+#                (\x60+[^\n\x60]+\x60+), \
+#                (\$+[^\n$]+\$+)
+
+# BlockIgnores = (?s) *(<code>.*?</code>)
+
```
Comment (ran-huang) [inline .vale.ini:14]:
```suggestion
```

File: styles/PingCAP/Gender.yml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+extends: existence
+message: "Don't use '%s' as a gender-neutral pronoun."
```
Comment (lilin90) [inline styles/PingCAP/Gender.yml:2]: Would it be necessary to tell users what word to use?
```suggestion
message: "Don't use '%s' as a gender-neutral pronoun. Instead, use the singular 'they'."
```

**Review by lilin90 — APPROVED** [review]
LGTM

**Pull request merged**